src/main/java/org/elasticsearch/index/query/BaseTermQueryBuilder.java

```diff
@@ -0,0 +1,245 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query;
+
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Streamable;
+import org.elasticsearch.common.lucene.BytesRefs;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public abstract class BaseTermQueryBuilder<QB extends BoostableQueryBuilder<QB>> extends BaseQueryBuilder implements Streamable, BoostableQueryBuilder<QB> {
+    
+    /** Name of field to match against. */
+    protected String fieldName;
+
+    /** Value to find matches for. */
+    protected Object value;
+
+    /** Query boost. */
+    protected float boost = 1.0f;
+
+    /** Name of the query. */
+    protected String queryName;
+
+    /**
+     * Constructs a new base term query.
+     *
+     * @param fieldName  The name of the field
+     * @param value The value of the term
+     */
+    public BaseTermQueryBuilder(String fieldName, String value) {
+        this(fieldName, (Object) value);
+    }
+
+    /**
+     * Constructs a new base term query.
+     *
+     * @param fieldName  The name of the field
+     * @param value The value of the term
+     */
+    @SuppressWarnings("boxing")
```

Comment:
I think it is :) especially given that we don't suppress this type of warnings anywhere in our codebase, not sure we should introduce this here. We tend to suppress stuff around generics mainly, in general warnings that show up with default IDE's setup (which IDE? :P )